app/src/main/java/com/lucas/medicaltools/service/MainRepository.kt

```diff
@@ -0,0 +1,50 @@
+package com.lucas.medicaltools.service
+
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.MainViewState
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import retrofit2.Response
+import java.util.*
+object MainRepository {
+
+    private fun mapMedicalTools(response: Response<List<MedicalTool>>): Observable<MainViewState> {
```

Comment:
Move mapping functions that returns a view state to a Interactor. 

Interactor is for logic of mapping a response to a view state for your Activity.

This repository will be needed in your interactor, as the repo handles the cache'ing vs fetching the data from the API.